src/app/components/GoogleSheetsInput.svelte

```diff
@@ -0,0 +1,172 @@
+   *  appropriately. After a sign-in, the API is called.
+   */
+  function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+      getValuesFromSpreadSheet();
+    } else {
+      handleAuthClick();
+      getValuesFromSpreadSheet();
+    }
+  }
+
+  /**
+   *  Sign in the user upon button click.
+   */
+  function handleAuthClick(event) {
+    gapi.auth2.getAuthInstance().signIn();
+  }
+
+  /**
+   *  Sign out the user upon button click.
+   */
+  function handleSignoutClick(event) {
+    gapi.auth2.getAuthInstance().signOut();
+  }
+
+  /**
+   * Gets the unique spreadsheet ID from a Google Sheet URL
+   *
+   */
+  function getSpreadsheetId() {
+    googleSheetsURL = document.getElementById("googleSheetsURL").value;
+    if (googleSheetsURL) {
+      return googleSheetsURL.split("/")[5];
+    } else {
+      error = `Error: The url ${googleSheetsURL} is not a valid Google Sheets URl. Please paste a valid one.`;
+    }
+  }
+
+  /**
+   * Read word and count form storage
+   * Sample: https://docs.google.com/spreadsheets/d/1lzHxoMWHpdGecby4d0y15AVf80csA7iNkaMMEL54Q7g/edit#gid=0
+   */
+  async function getValuesFromSpreadSheet() {
+    const wordListObject = [];
+    let spreadsheetId = getSpreadsheetId() || "";
+
+    if (error) {
+      return error;
+    }
+
+    gapi.client.sheets.spreadsheets.values
+      .get({
+        spreadsheetId: spreadsheetId,
+        range: "A2:B",
+      })
+      .then(
+        async function (response) {
+          var range = response.result;
+          if (range.values.length > 0) {
+            for (let i = 0; i < range.values.length; i++) {
+              let row = range.values[i];
+              let word = row[0];
+              let wordCount = row[1];
+              if (!wordCount) {
+                wordCount = 0;
+              }
+              wordListObject.push([word, wordCount]);
+            }
+            error = null;
+
+            await worker.readGoogleSheet(spreadsheetId, wordListObject);
+          } else {
+            error = "Error: No data found in the Google Sheet.";
+          }
+        },
+        function (response) {
+          error = "Error: " + response.result.error.message;
+        }
+      );
+  }
+</script>
+
+<style>
+  .google-sheets {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+  }
+
+  .error {
+    background-color: #f8d7db;
+    color: #400000;
+    padding: 5px 10px;
+  }
+</style>
+
+<!-- <pre id="content" style="white-space: pre-wrap;" /> -->
+<div class="google-sheets">
+  {#if error}
+    <p class:error>{error}</p>
+  {/if}
+  <h2>Or Get Values from a Google Sheet</h2>
+  <InputField
+    label="Google Sheets URL"
+    id="googleSheetsURL"
+    bind:value={googleSheetsURL}
+    fullWidth={true} />
+  <Button color="blue" onClick={handleClientLoad}>Read Values</Button>
+</div>
+<svelte:head>
+  <script async defer src="https://apis.google.com/js/api.js">
+  </script>
+</svelte:head>
```

Comment:
I would prefer `<svelte:head>` go above the main HTML content (above the `<div>`)